File: source/android-source/pilgrimunitysdk/src/main/java/com/foursquare/pilgrimunitysdk/PilgrimUnitySDK.java
package com.foursquare.pilgrimunitysdk;

import android.content.Context;
import android.support.annotation.NonNull;

import com.foursquare.pilgrim.PilgrimSdk;
import com.foursquare.pilgrim.PilgrimUserInfo;

@SuppressWarnings("unused")
public final class PilgrimUnitySDK {

    private PilgrimUnitySDK() {

    }

    public static void init(@NonNull Context context, @NonNull String consumerKey, @NonNull String consumerSecret) {
        PilgrimSdk.with(new PilgrimSdk.Builder(context).consumer(consumerKey, consumerSecret));
        restartIfPreviouslyStarted(context);
    }

    private static void restartIfPreviouslyStarted(@NonNull Context context) {
        if (Utils.isStarted(context)) {
            PilgrimSdk.start(context);
        }
    }

}

File: source/pilgrim-unity-sdk/Assets/PilgrimUnitySDK/Plugins/iOS/PilgrimUnitySDK.m
//
//  PilgrimUnitySDK.m
//  PilgrimUnitySDK
//
//  Copyright © 2019 Foursquare. All rights reserved.
//

#import "PilgrimUnitySDK.h"

#import <Pilgrim/Pilgrim.h>

@interface PilgrimUnitySDKDelegate : NSObject <FSQPPilgrimManagerDelegate>

@end

@implementation PilgrimUnitySDKDelegate

- (void)pilgrimManager:(nonnull FSQPPilgrimManager *)pilgrimManager handleVisit:(nonnull FSQPVisit *)visit {

}

@end

static PilgrimUnitySDKDelegate * delegate = nil;

@implementation PilgrimUnitySDK

+ (void)load {
    delegate = [[PilgrimUnitySDKDelegate alloc] init];
}

+ (void)initWithConsumerKey:(NSString *)consumerKey consumerSecret:(NSString *)consumerSecret {
    [FSQPPilgrimManager sharedManager].debugLogsEnabled = true;

    [[FSQPPilgrimManager sharedManager] configureWithConsumerKey:consumerKey
                                                          secret:consumerSecret
                                                        delegate:delegate
                                                      completion:nil];

    [self restartIfPreviouslyStarted];
}

+ (void)restartIfPreviouslyStarted {
    if ([[NSUserDefaults standardUserDefaults] boolForKey:@"Started"]) {
        [[FSQPPilgrimManager sharedManager] start];
    }
}

@end
